.github/workflows/nightly-test.yaml

```diff
@@ -63,3 +63,24 @@ jobs:
         if: ${{ failure() && github.event_name == 'pull_request' }}
         uses: mxschmitt/action-tmate@v3
         timeout-minutes: 10
+
+  test-strict-interfaces:
+    name: Strict Interfaces Test ${{ matrix.release }} ${{ matrix.arch }}
+    strategy:
+      matrix:
+        os: ["ubuntu:20.04"]
+        arch: ["amd64", "arm64"]
+        release: ["1.31/edge", "1.30/edge"]
```

Comment:
This will not be maintainable.
Please do something like this:
https://github.com/canonical/k8s-snap/blob/main/.github/workflows/nightly-test.yaml#L47

so that the channels are dynamically obtained:
https://github.com/canonical/k8s-snap/blob/main/tests/integration/tests/test_version_upgrades.py#L23

Bonus points if you refactor this logic into a reusable function.

Reply:
Baking this into Python, then we can integrate this nicely in the integration test like your version upgrades.